corehq/apps/hqwebapp/views.py

```diff
@@ -481,27 +484,26 @@ def debug_notify(request):
 
 @require_POST
 def jserror(request):
-    stack = request.POST.get('stack', None)
-    message = request.POST.get('message', None)
-    if stack:
-        cache_key = ' '.join(map(lambda l: l.strip(), stack.split('\n'))[:3])
-    else:
-        cache_key = message
-
-    count = ExponentialBackoff.increment(cache_key)
-    if not ExponentialBackoff.should_backoff(cache_key):
-        notify_js_exception(
-            request,
-            message=message,
-            details={
-                'filename': request.POST.get('filename', None),
-                'line': request.POST.get('line', None),
-                'page': request.POST.get('page', None),
-                'agent': request.META.get('HTTP_USER_AGENT', None),
-                'js_stack': stack,
-                'count': count,
-            }
-        )
+    agent = request.META.get('HTTP_USER_AGENT', None)
+    if agent:
+        parsed_agent = httpagentparser.detect(agent)
+        os = browser_name = browser_version = DATADOG_UNKNOWN
+        bot = parsed_agent.get('bot', False)
+        if 'os' in parsed_agent:
+            os = parsed_agent['os'].get('name', DATADOG_UNKNOWN)
+
+        if 'browser' in parsed_agent:
+            browser_version = parsed_agent['browser'].get('version', DATADOG_UNKNOWN)
+            browser_name = parsed_agent['browser'].get('name', DATADOG_UNKNOWN)
```

Comment:
Is the nesting wrong maybe? Looks like if agent is falsy, you'll get down to the `log_counter` and `os`, `browser_version`, `browser_name`, and `bot` won't be defined.